### PR TITLE
Add PAMPA toolsuite

### PR DESCRIPTION
**Pull request by ColineRoyaux**
Tools to compute and analyse biodiversity metrics based on the Essential Biodiversity Variables concept and the tcltk R tool PAMPA (https://wwz.ifremer.fr/pampa/Meth.-Outils/Outils)

### CODE REVIEW

**Comment by ColineRoyaux** [comment]
@bgruening Sorry for the omission ! Thank you very much

File: tools/PAMPA/PAMPA_GLM_SP.xml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        <param name="input_unitobs" type="data" format="tabular" label="Unitobs informations file" help="Unitobs file, with all informations available about unitobs."/>
+        <param name="varint" type="data_column" data_ref="input_metric" label="Interest variable from population metrics file" help= "Choose the field of the interest variable."/>
+        <param name="varrep" type="select" label="Response variables" help= "Choose the response variables you want to include in your analysis." multiple="true">
+            <option selected="true" value="year">Year</option>
+            <option selected="true" value="site">Site</option>
+            <option selected="true" value="habitat1">Habitat</option>
+        </param>
+        <param name="varrand" type="select" label="Random effect ?" help="Allocate a random effect on site or year makes your model more reliable as random events on a peculiar site or year can affect populations, it takes account of pseudoreplication. However, avoid applying it on a less than 10 levels variable (less than 10 different sites and/or year)." multiple="true">
```
Comment (bgruening) [inline tools/PAMPA/PAMPA_GLM_SP.xml:38]: this could be in the maro

File: tools/PAMPA/PAMPA_GLM_SP.xml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+         '$__tool_directory__/FunctPAMPAGalaxy.r' 
+         '$output_summary'
+         '$output_recap'
+    ]]>
+    </command>
+    <inputs>
+        <param name="input_metric" type="data" format="tabular" label="Input population metrics file" help="Metrics data file, with location, year, and metrics informations as 'presence absence', 'abundance' or others who can be used as interest variable."/>
+        <param name="input_unitobs" type="data" format="tabular" label="Unitobs informations file" help="Unitobs file, with all informations available about unitobs."/>
+        <param name="varint" type="data_column" data_ref="input_metric" label="Interest variable from population metrics file" help= "Choose the field of the interest variable."/>
+        <param name="varrep" type="select" label="Response variables" help= "Choose the response variables you want to include in your analysis." multiple="true">
```
Comment (bgruening) [inline tools/PAMPA/PAMPA_GLM_SP.xml:33]: that could also be in the macros, most of this tool is shared with the other tool above.

File: tools/PAMPA/PAMPA_GLM_SP.xml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            </when>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data name="output_summary" from_work_dir="GLMSummary.txt" format="txt" label="GLM - Results from your population analysis on ${on_string}"/>
+        <data name="output_recap" from_work_dir="GLMSummaryFull.txt" format="txt" label="Simple statistics on chosen variables on ${on_string}"/>
+    </outputs>
+    <tests>
+        <test> 
+
```
Comment (bgruening) [inline tools/PAMPA/PAMPA_GLM_SP.xml:65]: tests are missing

File: tools/PAMPA/PAMPA_GLM.xml [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            </when>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data name="output_summary" from_work_dir="GLMSummary.txt" format="txt" label="GLM - Results from your Analysis on ${on_string}"/>
+        <data name="output_recap" from_work_dir="GLMSummaryFull.txt" format="txt" label="Simple statistics on chosen variables on ${on_string}"/>
+    </outputs>
+    <tests>
+        <test> 
+
```
Comment (bgruening) [inline tools/PAMPA/PAMPA_GLM.xml:66]: tests are missing

File: tools/PAMPA/CalculatePresAbs.xml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    </command>
+    <inputs>
+        <expand macro="pampa_input_calculate"/>
+    </inputs>
+    <outputs>
+        <data name="output_presabs" from_work_dir="TabPresAbs.tabular" format="tabular"/>
+    </outputs>
+    <tests>
+        <test> 
+
```
Comment (bgruening) [inline tools/PAMPA/CalculatePresAbs.xml:23]: tests are missing

File: tools/PAMPA/CalculateCommunityMetrics.xml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            <option value="simpson.l">1-Simpson</option>
+            <option value="pielou">Pielou</option>
+            <option value="hill">Hill</option>
+        </param>
+    </inputs>
+    <outputs>
+        <data name="output_community" from_work_dir="TabCommunityIndexes.tabular" format="tabular"/>
+    </outputs>
+    <tests>
+        <test>
```
Comment (bgruening) [inline tools/PAMPA/CalculateCommunityMetrics.xml:29]: tests are missing

**Comment by yvanlebras** [comment]
@ColineRoyaux Tests data input/output and parameters are ok ?

**Comment by ColineRoyaux** [comment]
@bgruening Thanks a lot for your review ! I think I made all the corrections you suggested and all tests are passed on my computer tell me if ever I missed something

**Comment by yvanlebras** [comment]
Fine for me! Amazing job Coline, thank you Björn!

**Comment by bgruening** [comment]
Cool, @yvanlebras do you add this to EU?

**Comment by yvanlebras** [comment]
Of course (if I remember the different places to add it ;) )

**Comment by bgruening** [comment]
@ColineRoyaux @yvanlebras the tools have never been deployed. https://travis-ci.org/github/galaxyecology/tools-ecology

I think the tests failed because of rounding errors. Maybe you assert for the tests. e.g. lline-count assert, regex, #columns

**Comment by yvanlebras** [comment]
Thank you Björn ! And Coline, as it appears now this is fixed: https://travis-ci.org/github/galaxyecology/tools-ecology

**Comment by yvanlebras** [comment]
In fact, it seems test and lint are ok, but not deploy... Any idea @bgruening ?

**Comment by bgruening** [comment]
Yes, the TTS is down :( as a workaround uncomment the TTS deployment, make a whitespace change in one of this tools here and commit again.

**Pull request merged**